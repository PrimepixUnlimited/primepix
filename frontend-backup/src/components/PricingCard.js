import React from 'react';
import {PricingCard} from 'react-native-elements';

const PricingCardWrapper = ({
  button,
  color,
  containerStyle,
  info,
  onButtonPress,
  price,
  title,
}) => (
  <PricingCard
    button={button}
    color={color}
    containerStyle={containerStyle}
    info={info}
    onButtonPress={onButtonPress}
    price={price}
    title={title}
  />
);

export default PricingCardWrapper;
