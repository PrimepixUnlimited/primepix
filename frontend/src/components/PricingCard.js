import React from 'react';
import {PricingCard} from 'react-native-elements';

const PricingCardWrapper = ({
  button,
  color,
  info,
  onButtonPress,
  price,
  title,
}) => (
  <PricingCard
    button={button}
    color={color}
    info={info}
    onButtonPress={onButtonPress}
    price={price}
    title={title}
  />
);

export default PricingCardWrapper;
