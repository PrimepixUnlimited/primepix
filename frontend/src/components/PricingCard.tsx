import React, { FC, ReactNode } from 'react'
import { StyleSheet, View } from 'react-native'
import { Text } from 'react-native-elements'

import Badge from './Badge'

import styles from '../constants/styles'

interface Props {
  button?: ReactNode
  currency: string
  discount?: number
  info: {
    title: string
    active: boolean
  }[]
  isActive?: boolean
  period?: string
  price: string
  title: string
}

const PricingCardWrapper: FC<Props> = ({
  button,
  currency,
  discount,
  info,
  isActive,
  period,
  price,
  title
}) => (
  <View style={s.container}>
    <View style={s.headingContainer}>
      {isActive && (
        <Badge
          status="success"
          containerStyle={{ position: 'absolute', top: -4, right: -4 }}
        />
      )}
      <Text style={s.headingText}>{title}</Text>
    </View>
    <View style={{ padding: 20, alignItems: 'center' }}>
      <View style={s.priceContainer}>
        <Text style={s.currencyText}>{currency}</Text>
        <Text style={s.priceText}>{price}</Text>
        <Text style={[s.currencyText, { alignSelf: 'flex-end' }]}>.00</Text>
      </View>
      {discount && <Text style={s.discountText}>Save up to {discount}%</Text>}
      <View style={{ marginVertical: 25 }}>
        {info.map((feature, idx) => (
          <Text
            key={idx}
            style={[s.featureText, style(feature.active).featureText]}
          >
            {feature.title}
          </Text>
        ))}
      </View>
      {button}
    </View>
  </View>
)

const s = StyleSheet.create({
  container: {
    backgroundColor: styles.greyScale.black2,
    marginBottom: 20
  },
  headingContainer: {
    backgroundColor: styles.greyScaleOpacity(0.05).white,
    alignItems: 'center',
    width: '100%'
  },
  headingText: {
    color: styles.greyScale.white,
    fontSize: 21,
    lineHeight: 35,
    fontWeight: 'bold',
    padding: 20
  },
  currencyText: {
    color: styles.greyScale.white,
    fontSize: 24,
    marginTop: 8
  },
  priceContainer: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    height: 56,
    marginBottom: 10
  },
  priceText: {
    color: styles.colors.success,
    fontSize: 52
  },
  periodText: {
    color: styles.greyScaleOpacity(0.5).white,
    fontSize: 16,
    lineHeight: 20,
    marginBottom: 25
  },
  discountText: {
    color: styles.greyScaleOpacity(0.5).white,
    fontSize: 16,
    lineHeight: 20
  },
  featureText: {
    textAlign: 'center',
    lineHeight: 20,
    paddingVertical: 5
  }
})

const style = (active: boolean) =>
  StyleSheet.create({
    featureText: {
      color: active
        ? styles.greyScale.white
        : styles.greyScaleOpacity(0.5).white,
      textDecorationLine: active ? 'none' : 'line-through'
    }
  })

export default PricingCardWrapper
