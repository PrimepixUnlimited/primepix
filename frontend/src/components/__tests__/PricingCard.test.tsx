// components/__tests__/Hello.tsx
import 'react-native'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import PricingCard from '../PricingCard'

const info = [
  {
    title: '5 Products',
    active: true
  },
  {
    title: 'Payout once a month',
    active: true
  },
  {
    title: 'Full Support',
    active: false
  },
  {
    title: 'All Core Features',
    active: true
  }
]

describe('PricingCard snapshot', () => {
  it('renders correctly', () => {
    const pricingCard = renderer
      .create(<PricingCard currency="Â£" info={info} price="30" title="Basic" />)
      .toJSON()
    expect(pricingCard).toMatchSnapshot()
  })
})
