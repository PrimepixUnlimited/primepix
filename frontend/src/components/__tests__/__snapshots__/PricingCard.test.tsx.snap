// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingCard snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#202124",
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.05)",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 21,
          "fontWeight": "bold",
          "lineHeight": 35,
          "padding": 20,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Basic
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "height": 56,
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 24,
            "marginTop": 8,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Â£
      </Text>
      <Text
        style={
          Object {
            "color": "#0bd685",
            "fontSize": 52,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        30
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#fff",
            "fontSize": 24,
            "marginTop": 8,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        .00
      </Text>
    </View>
    <View
      style={
        Object {
          "marginVertical": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "lineHeight": 20,
            "paddingVertical": 5,
            "textAlign": "center",
            "textDecorationLine": "none",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        5 Products
      </Text>
      <Text
        style={
          Object {
            "color": "#fff",
            "lineHeight": 20,
            "paddingVertical": 5,
            "textAlign": "center",
            "textDecorationLine": "none",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Payout once a month
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.5)",
            "lineHeight": 20,
            "paddingVertical": 5,
            "textAlign": "center",
            "textDecorationLine": "line-through",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Full Support
      </Text>
      <Text
        style={
          Object {
            "color": "#fff",
            "lineHeight": 20,
            "paddingVertical": 5,
            "textAlign": "center",
            "textDecorationLine": "none",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        All Core Features
      </Text>
    </View>
  </View>
</View>
`;
